### PR TITLE
Feature/content discover  search page [Part 1]

### PR DESCRIPTION
**Pull request by lrojas94**

### CODE REVIEW

File: src/components/Search/CustomSearchComponents.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+export function SearchableMultiSelectRefinementList({
+  title,
+  ...props
+}: RefinementListProps & { title: string }) {
+  const { items, refine, searchForItems, isFromSearch } = useRefinementList({ ...props });
+  const [searchValue, setSearchValue] = useState('');
+  const [debouncedSearchValue] = useDebouncedValue(searchValue, 300);
+  // We need to keep the state of the select here because the items may dissapear while searching.
+  const [refinedItems, setRefinedItems] = useState<typeof items>([]);
```
Comment (manuelurenah) [inline src/components/Search/CustomSearchComponents.tsx:47]: Wouldn't it be good to initialize this with the `items`?
```suggestion
  const [refinedItems, setRefinedItems] = useState<typeof items>(items ?? []);
```
Not sure if items is null at some point, so doing fallback to empty array

File: src/components/Search/CustomSearchComponents.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+  const onUpdateSelection = (updatedSelectedItems: string[]) => {
+    const addedItems = updatedSelectedItems.length > refinedItems.length;
+    if (addedItems) {
+      // Get the last item:
+      const lastAddedValue = updatedSelectedItems[updatedSelectedItems.length - 1];
```
Comment (manuelurenah) [inline src/components/Search/CustomSearchComponents.tsx:53]: Nitpick 🤓 
```suggestion
      const lastAddedValue = updatedSelectedItems.at(-1);
```

File: src/components/Search/SearchHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  SegmentedControlItem,
+  Text,
+  ThemeIcon,
+  Title,
+} from '@mantine/core';
+import { IconCategory, IconFileText } from '@tabler/icons-react';
+import { useRouter } from 'next/router';
+import { removeEmpty } from '~/utils/object-helpers';
+
+const useStyles = createStyles((theme) => ({}));
```
Comment (manuelurenah) [inline src/components/Search/SearchHeader.tsx:16]: Seems like we are only using this to get the theme, we can replace it with `useMantineTheme`
```suggestion
```

File: src/components/Search/SearchHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+export const SearchHeader = () => {
+  const { uiState } = useInstantSearch();
+  const { setSearchParamsByUiState, ...states } = useSearchStore((state) => state);
+  const { theme } = useStyles();
```
Comment (manuelurenah) [inline src/components/Search/SearchHeader.tsx:21]:
```suggestion
  const theme = useMantineTheme();
```

File: src/components/Search/SearchHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  };
+
+  const data: SegmentedControlItem[] = [
+    {
+      label: (
+        <Group align="center" spacing={8} noWrap>
+          <ThemeIcon
+            size={30}
+            color={index === 'models' ? theme.colors.dark[7] : 'transparent'}
+            p={6}
```
Comment (manuelurenah) [inline src/components/Search/SearchHeader.tsx:54]: Based on the designs, I believe these should be rounded
```suggestion
            p={6}
            radius="xl"
```

File: src/components/Search/SearchHeader.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+      ),
+      value: 'models',
+    },
+    {
+      label: (
+        <Group align="center" spacing={8} noWrap>
+          <ThemeIcon
+            size={30}
+            color={index === 'articles' ? theme.colors.dark[7] : 'transparent'}
+            p={6}
```
Comment (manuelurenah) [inline src/components/Search/SearchHeader.tsx:71]: Ditto

File: src/components/Search/useSearchState.ts [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+import singletonRouter from 'next/router';
+import { InstantSearchProps } from 'react-instantsearch-hooks-web';
+import { createInstantSearchRouterNext } from 'react-instantsearch-router-nextjs';
+
+export const searchIndexes = ['models', 'articles'] as const;
+export type SearchIndex = (typeof searchIndexes)[number];
+export const searchParamsSchema = z.object({
+  query: z
+    .any()
+    .transform((val) => val.toString())
```
Comment (manuelurenah) [inline src/components/Search/useSearchState.ts:17]: Try this one
```suggestion
  query: z.coerce.string()
```

File: src/pages/search/articles.tsx [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+            padding: theme.spacing.md,
+          })}
+        >
+          <RenderFilters />
+        </Stack>
+
+        <Stack pl={377} w="100%">
+          <SearchHeader />
+          <ArticlesHitList />
+        </Stack>
```
Comment (manuelurenah) [inline src/pages/search/articles.tsx:47]: Wouldn't it make sense to use `<Grid>` in here to maintain consistency between breakpoints? Having fixed width feels weird

File: src/pages/search/models.tsx [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+            padding: theme.spacing.md,
+          })}
+        >
+          <RenderFilters />
+        </Stack>
+
+        <Stack pl={377} w="100%">
+          <SearchHeader />
+          <ModelsHitList />
+        </Stack>
```
Comment (manuelurenah) [inline src/pages/search/models.tsx:64]: Same as before, maybe using `<Grid>` works best here

File: src/pages/search/models.tsx [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+  return (
+    <>
+      <SortBy
+        title="Sort models by"
+        items={[
+          { label: 'Highest Rated', value: 'models:metrics.weightedRating:desc' },
+          { label: 'Most downloaded', value: 'models:metrics.downloadCount:desc' },
+          { label: 'Most Liked', value: 'models:metrics.favoriteCount:desc' },
+          { label: 'Most discussed', value: 'models:metrics.commentCount:desc' },
+          { label: 'Newest', value: 'models:createdAt:desc' },
```
Comment (manuelurenah) [inline src/pages/search/models.tsx:80]: Let's keep word casing consistency in the labels 😛 
```suggestion
          { label: 'Highest rated', value: 'models:metrics.weightedRating:desc' },
          { label: 'Most downloaded', value: 'models:metrics.downloadCount:desc' },
          { label: 'Most liked', value: 'models:metrics.favoriteCount:desc' },
          { label: 'Most discussed', value: 'models:metrics.commentCount:desc' },
          { label: 'Newest', value: 'models:createdAt:desc' },
```

File: src/pages/search/articles.tsx [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+const RenderFilters = () => {
+  return (
+    <>
+      <SortBy
+        title="Sort models by"
+        items={[
+          { label: 'Most Bookmarked', value: 'articles:metrics.favoriteCount:desc' },
+          { label: 'Most Viewed', value: 'articles:metrics.viewCount:desc' },
+          { label: 'Most Discussed', value: 'articles:metrics.commentCount:desc' },
+          { label: 'Newest', value: 'articles:createdAt:desc' },
```
Comment (manuelurenah) [inline src/pages/search/articles.tsx:62]: From the comment down below, let's make the labels casing consistent
```suggestion
          { label: 'Most bookmarked', value: 'articles:metrics.favoriteCount:desc' },
          { label: 'Most viewed', value: 'articles:metrics.viewCount:desc' },
          { label: 'Most discussed', value: 'articles:metrics.commentCount:desc' },
          { label: 'Newest', value: 'articles:createdAt:desc' },
```

File: src/pages/search/models.tsx [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+      showMore?.();
+    }
+  }, [status, inView, showMore, isLastPage]);
+
+  return (
+    <Stack>
+      <Box className={classes.grid}>
+        {hits.map((hit) => {
+          const modelHit = hit as unknown as ModelGetAll['items'][number];
+          const images = (hit.images ?? []) as ModelGetAll['items'][number]['image'][];
```
Comment (manuelurenah) [inline src/pages/search/models.tsx:139]: We can avoid these castings if we explicitly set the generic in `useInfiniteHits`

File: src/pages/search/models.tsx [hunk 13]
```diff
@@ -0,0 +1,156 @@
...
+    marginTop: -theme.spacing.md,
+
+    '& > *': {
+      marginTop: theme.spacing.md,
+    },
+  },
+}));
+
+export function ModelsHitList() {
+  const { hits, showMore, isLastPage } = useInfiniteHits();
```
Comment (manuelurenah) [inline src/pages/search/models.tsx:122]: We now have a type coming from the search index 🙂 just make sure to get the import
```suggestion
  const { hits, showMore, isLastPage } = useInfiniteHits<ModelSearchIndexRecord>();
```

File: src/components/Search/CustomSearchComponents.tsx (hunk 2 above)
Comment (lrojas94) [reply src/components/Search/CustomSearchComponents.tsx:53]: Breaks safari 😬

File: src/components/Search/CustomSearchComponents.tsx (hunk 1 above)
Comment (lrojas94) [reply src/components/Search/CustomSearchComponents.tsx:47]: Items is not the same as refined items tho

File: src/pages/search/articles.tsx (hunk 8 above)
Comment (lrojas94) [reply src/pages/search/articles.tsx:47]: Yeah some improvements to be there, but I don't think Grid is the right call here. 
Technically, we do want a fixed grid for the filters. And we should move the items to the side that amount except for mobile.

**Pull request merged**